web/src/compose_validate.ts

```diff
@@ -329,6 +330,49 @@ export function warn_if_in_search_view(): void {
     }
 }
 
+export function clear_guest_in_dm_recipient_warning(): void {
+    const classname = compose_banner.CLASSNAMES.guest_in_direct_message_recipient;
+    $(`#compose_banners .${CSS.escape(classname)}`).remove();
+}
+
+// Only called on recipient change. Adds new banner if not already
+// exists or updates the existing banner or remove banner is no
+// guest in the dm.
+export function warn_if_guest_in_dm_recipient(): void {
+    const email_string = compose_state.private_message_recipient();
+    const guest_emails = people.filter_guest_emails(email_string);
+
+    // Only messages of type private are valid
+    const is_valid = compose_state.get_message_type() === "private";
+
+    if (!realm.realm_enable_guest_user_dm_warning || !is_valid || !guest_emails) {
+        clear_guest_in_dm_recipient_warning();
+        return;
+    }
+
+    const classname = compose_banner.CLASSNAMES.guest_in_direct_message_recipient;
+    let $banner = $(`#compose_banners .${CSS.escape(classname)}`);
+    const banner_text = people.get_dm_guest_banner_text(guest_emails);
+
+    if (compose_state.get_recipient_guest_emails() === guest_emails.join(",")) {
+        return;
+    }
```

Comment:
It would be good to move this early return before all the variables are declared, I was wondering why none of the variables declared just before this code block are not used here.